src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinInetProxyHelper.cs

```diff
@@ -188,16 +188,14 @@ public bool AutoDetectScriptSuccess()
             if (!AutoDetect) return false;
 
             bool doAutoDetect = false;
-            object executionLock = new object();
-            long currentTimeStamp = DateTime.Now.Ticks;
-
-            lock (executionLock)
-            {
-                doAutoDetect = _previousTimeStampTicks == DateTime.MinValue.Ticks ?
-                    true : (currentTimeStamp - _previousTimeStampTicks > TimerPeriod ? true : false);
-
-                _previousTimeStampTicks = currentTimeStamp;
-            }
+            int currentTimeStamp = Environment.TickCount;
+
+            // Environment.TickCount will increment from zero to Int32.MaxValue for approximately 24.9 days, then jump to
+            // Int32.MinValue. When that happens, we may skip doing WinHttpDetectAutoProxyConfigUrl for one request which we
```

Comment:
Why would we skip at that point?  e.g. this will output 1:
```C#
int a = int.MaxValue;
int b = int.MinValue;
Console.WriteLine(b - a);
```
I think the only time we'd end up skipping with this logic is if two requests were more than 48 days apart.